tools/audio/extraction/audiotable.py

```diff
@@ -0,0 +1,690 @@
+
+    def clone(self, start, end, padding):
+        new_sample = AudioTableSample(start, end, self.header, self.data, self.book, self.loop, padding)
+        new_sample.notes_rates = self.notes_rates
+        return new_sample
+
+    def frame_size(self):
+        return {
+            AudioSampleCodec.CODEC_ADPCM        : 9,
+            AudioSampleCodec.CODEC_S8           : 16,
+            AudioSampleCodec.CODEC_S16_INMEMORY : 32,
+            AudioSampleCodec.CODEC_SMALL_ADPCM  : 5,
+            AudioSampleCodec.CODEC_REVERB       : 0,
+            AudioSampleCodec.CODEC_S16          : 32
+        }[self.header.codec]
+
+    def codec_id(self):
+        return {
+            AudioSampleCodec.CODEC_ADPCM        : b'ADP9',
+            AudioSampleCodec.CODEC_S8           : b'HPCM',
+            AudioSampleCodec.CODEC_S16_INMEMORY : b'NONE',
+            AudioSampleCodec.CODEC_SMALL_ADPCM  : b'ADP5',
+            AudioSampleCodec.CODEC_REVERB       : b'RVRB',
+            AudioSampleCodec.CODEC_S16          : b'NONE',
+        }[self.header.codec]
+
+    def codec_name(self):
+        return {
+            AudioSampleCodec.CODEC_ADPCM        : b"Nintendo/SGI VADPCM 9-bytes/frame",
+            AudioSampleCodec.CODEC_S8           : b"Half-frame PCM",
+            AudioSampleCodec.CODEC_S16_INMEMORY : b"Uncompressed",
+            AudioSampleCodec.CODEC_SMALL_ADPCM  : b"Nintendo/SGI VADPCM 5-bytes/frame",
+            AudioSampleCodec.CODEC_REVERB       : b"Nintendo Reverb format",
+            AudioSampleCodec.CODEC_S16          : b"Uncompressed"
+        }[self.header.codec]
+
+    def codec_file_extension_compressed(self):
+        ext = {
+            AudioSampleCodec.CODEC_ADPCM        : ".aifc",
+            AudioSampleCodec.CODEC_S8           : None,
+            AudioSampleCodec.CODEC_S16_INMEMORY : None,
+            AudioSampleCodec.CODEC_SMALL_ADPCM  : ".half.aifc",
+            AudioSampleCodec.CODEC_REVERB       : None,
+            AudioSampleCodec.CODEC_S16          : ".aiff",
+        }[self.header.codec]
+        assert ext is not None
+        return ext
+
+    def codec_file_extension_decompressed(self):
+        ext = {
+            AudioSampleCodec.CODEC_ADPCM        : ".wav",
+            AudioSampleCodec.CODEC_S8           : None,
+            AudioSampleCodec.CODEC_S16_INMEMORY : None,
+            AudioSampleCodec.CODEC_SMALL_ADPCM  : ".half.wav",
+            AudioSampleCodec.CODEC_REVERB       : None,
+            AudioSampleCodec.CODEC_S16          : ".wav",
+        }[self.header.codec]
+        assert ext is not None
+        return ext
+
+    def base_note_number(self):
+        return note_z64_to_midi(pitch_names.index(self.base_note))
+
+    def resolve_basenote_rate(self, extraction_sample_info : Dict[int, Dict[str,str]]):
+        assert len(self.notes_rates) != 0
+
+        # rate_3ds = None
+        # if SAMPLERATES_3DS is not None:
+        #     rate_3ds = SAMPLERATES_3DS[self.bank_num].get(i, None)
```

Comment:
Still some dead cod

Reply:
Not removing this code in this function is intended, in case there's any interest in reviving 3DS comparisons. I felt it made the function harder to read if I uncommented it though.

ditto for everything else

Reply:
It doesn't really feel worth adding more to this, it's pretty clear from reading the comments that it's for 3DS samplerate comparisons

Reply:
Perhaps it is just me, but I find dead inherently more confusion prone, which is why I pointed it out. It always makes me second guess it's purpose, and also if it is fully complete or is there something else I need to uncomment out or implement to get it to work fully. However nobody else seems to have had issues with it, so if you think it is enough I'll concede.